packages/ui/react-ui-table/src/components/Table/props.ts

```diff
@@ -0,0 +1,50 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { type RowData, type RowSelectionState, type Table, type VisibilityState } from '@tanstack/react-table';
+
+import { type KeyValue, type TableColumnDef } from '../../types';
+
+export type TableFlags = Partial<{
+  role: 'table' | 'grid' | 'treegrid';
+  grouping: string[];
+  header: boolean;
+  footer: boolean;
+  border: boolean;
+  fullWidth: boolean;
+  debug: boolean;
+  expand: boolean;
+  rowsSelectable: boolean | 'multi';
+}>;
+
+export type TableCurrent<TData extends RowData> = Partial<{
```

Comment:
Assume this is "current" in the "aria-current" sense?
Would appreciate light sprinkles of comments for the main Table components/props/concepts.